onnxscript/ir/_external_data.py

```diff
@@ -51,3 +70,213 @@ def set_base_dir(graph: _core.Graph | _core.GraphView, base_dir: str | os.PathLi
+        model.graph.initializers[value.name] = value
+    return model
+
+
+def to_external_data(
+    model: _core.Model,
+    base_path: str | os.PathLike,
+    file_path: str = "",
+    align_offset: bool = True,
+    align_threshold: int = 1048576,  # 1MB,
+    allocation_granularity: int = 65536,  # 64KB
+) -> _core.Model:
+    """
+    Call to set all tensors with raw data as external data.
+
+    Args:
+        model: Model to be converted.
+        base_path: Path of base directory.
+        file_path: Path to which external data is to be stored.
+        align_offset: Offset will always be page aligned and alloction granularity aligned for mmap support. This is done by padding previous tensor data with zeros keeping same length. Tensor data will be aligned if > align_threshold
+        align_threshold: Alignment threshold for size of data. Having a low threshold will waste file space for small initializers. Only when tensor's data is > the page_align_threshold it will be force aligned.
+        allocation_granularity: The allocation Granularity for mmap() support. Typically 64KB for Windows & 4KB for other OSes.
+    """
+    file_path = pathlib.Path(base_path) / file_path
+    # Check if file path is valid, and create subsequent subdirectories within the path if they don't exist
+    try:
+        file_path.mkdir(parents=True, exist_ok=True)
+    except:
+        pass
+    # Check if file is empty. If not, load pre-existing external data.
+    check_external_data_file(model, file_path)
+
+    # Get all the tensors in the graph which are to be stored as external data.
+    # Iterate through all the tensors, and extract the external data information such as
+    # name, offset and length.
+    # TODO: Currently attributes not handled, eventually try to use _all_tensors to include attrs
+    tensors: list[tuple[_core.Value, _protocols.TensorProtocol]] = []
+    for value in model.graph.initializers.values():
+        if value.const_value is not None:
+            tensors.append((value, value.const_value))
+    external_data_info: list[tuple[_core.Value, ExternalDataInfo]] = []
+
+    # Sort all tensors based on tensor sizes, in order to avoid unneccesarry alignment.
+    # All the smaller tensors are written earlier and alignment is peformed for the larger tensors.
+    tensors = sorted(tensors, key=lambda x: x[1].nbytes)
+
+    current_offset = 0
+    for value, tensor in tensors:
+        tensor_info = set_external_data(
+            tensor,
+            current_offset,
+            align_offset=align_offset,
+            align_threshold=align_threshold,
+            allocation_granularity=allocation_granularity,
+        )
+        external_data_info.append((value, tensor_info))
+        current_offset = tensor_info.offset + tensor_info.length
+
+    save_external_data(external_data_info, file_path)
+
+    # Convert initializers to ExternalTensors
+    model = store_as_external_tensors(model, external_data_info, file_path)
```

Comment:
```suggestion
    model = _store_as_external_tensors(model, external_data_info, file_path)
```

I would make all internal functions private